engine/access/rpc/backend/scripts/executor/local.go

```
@@ -2,112 +2,170 @@ package executor
+//   - [access.InternalError] - for internal failures or index conversion errors.
+func convertScriptExecutionError(err error) error {
+	switch {
+	case errors.Is(err, version.ErrOutOfRange),
+		errors.Is(err, storage.ErrHeightNotIndexed),
+		errors.Is(err, execution.ErrIncompatibleNodeVersion):
+		return access.NewOutOfRangeError(err)
+	case errors.Is(err, storage.ErrNotFound):
+		return access.NewDataNotFoundError("header", err)
 	}
 
 	var failure fvmerrors.CodedFailure
 	if fvmerrors.As(err, &failure) {
-		return rpc.ConvertError(err, "failed to execute script", codes.Internal)
+		return access.NewInternalError(err)
 	}
 
 	// general FVM/ledger errors
 	var coded fvmerrors.CodedError
 	if fvmerrors.As(err, &coded) {
 		switch coded.Code() {
 		case fvmerrors.ErrCodeScriptExecutionCancelledError:
-			return status.Errorf(codes.Canceled, "script execution canceled: %v", err)
-
+			return access.NewRequestCanceledError(err)
 		case fvmerrors.ErrCodeScriptExecutionTimedOutError:
-			return status.Errorf(codes.DeadlineExceeded, "script execution timed out: %v", err)
-
+			return access.NewRequestTimedOutError(err)
 		case fvmerrors.ErrCodeComputationLimitExceededError:
-			return status.Errorf(codes.ResourceExhausted, "script execution computation limit exceeded: %v", err)
-
+			err = fmt.Errorf("script execution computation limit exceeded: %w", err)
+			return access.NewResourceExhausted(err)
 		case fvmerrors.ErrCodeMemoryLimitExceededError:
-			return status.Errorf(codes.ResourceExhausted, "script execution memory limit exceeded: %v", err)
-
+			err = fmt.Errorf("script execution memory limit exceeded: %w", err)
+			return access.NewResourceExhausted(err)
 		default:
 			// runtime errors
-			return status.Errorf(codes.InvalidArgument, "failed to execute script: %v", err)
+			return access.NewInvalidRequestError(err)
 		}
 	}
 
-	return rpc.ConvertIndexError(err, height, "failed to execute script")
+	return access.NewInternalError(err)
```

this is catching all unexpected errors, and converting them to expected errors. This should instead just return the error directly.